apps/app_chanspy.c

```diff
@@ -471,6 +478,46 @@ static int spy_generate(struct ast_channel *chan, void *data, int len, int sampl
 	if (ast_test_flag(&csth->flags, OPTION_READONLY)) {
 		/* Option 'o' was set, so don't mix channel audio */
 		f = ast_audiohook_read_frame(&csth->spy_audiohook, samples, AST_AUDIOHOOK_DIRECTION_READ, ast_format_slin);
+	} else if (ast_test_flag(&csth->flags, OPTION_INTERLEAVED)) {
+		/* Option 'D' was set, so mix the spy frame as an interleaved dual channel frame. */
+		struct ast_frame *fr_read = NULL;
+		struct ast_frame *fr_write = NULL;
+		short stereo_buf[samples * 2];
+		struct ast_frame stereo_frame = {
+			.frametype = AST_FRAME_VOICE,
+			.datalen = sizeof(stereo_buf),
+			.samples = samples,
+		};
+		short read_buf[samples];
+		short write_buf[samples];
+		f = ast_audiohook_read_frame_all(&csth->spy_audiohook, samples, ast_format_slin, &fr_read, &fr_write);
+		if (f) {
+			ast_frame_free(f, 0);
+			f = NULL;
+		}
+		memset(read_buf, 0, sizeof(read_buf));
+		memset(write_buf, 0, sizeof(write_buf));
+		if (fr_read) {
+			memcpy(read_buf, fr_read->data.ptr, sizeof(read_buf));
```

Comment:
Why memset right before we memcpy?
Would make more sense to have:
if fr_read memcpy else memset